src/universal/modules/landing/containers/Landing/LandingContainer.js

```diff
@@ -42,6 +42,8 @@ query {
 const mutationHandlers = {
   updateUserWithAuthToken(optimisticVariables, dataFromServer, currentResponse) {
     if (dataFromServer) {
+      // TODO: modifing params by reference is stylisitically problematic,
+      //       can interface be pure and return new value intead?
```

Comment:
I think it's always good to be aware of mutating objects, but it shouldn't be a hard rule.
For example (since i learned about plugins yesterday...) almost every webpack plugin does this because it hands you a `compilation` object and you stick something on it, eg `compilation.assets[myFoo] = 'newFoo!'`. What else are they gonna do, hand you the parent of `compilation` and make you return a whole new `compilation` object? Nahhhh

Cashay does _allow_ you to return a pure object, but here's how that looks:

Option 1:
```
const myResponse = JSON.parse(JSON.stringify(currentResponse));
myResponse.foo = 2;
return myResponse;
```
Option 2:
```
return Object.assign({}, currentResponse, {
  cachedUserAndToken: Object.assign({}, currentResponse.cachedUserAndToken, {
    user: Object.assign({}, currentResponse.cachedUserAndToken.user, {
      id: dataFromServer.cachedUserAndToken.user.id
    }
  })
}
```

So I guess the answer is, want to play by the rules? Go ahead, no one is stopping you. But, if you wanna be a bad boy (without any risk!), go ahead & mutate 😈